src/providers/WorkflowCore.Persistence.MongoDB/Services/MongoPersistenceProvider.cs

```diff
@@ -24,6 +26,13 @@ public MongoPersistenceProvider(IMongoDatabase database)
 
         static MongoPersistenceProvider()
         {
+            ConventionRegistry.Register(
+                "workflow.conventions",
+                new ConventionPack
+                {
+                    new EnumRepresentationConvention(BsonType.String)
+                }, t => t.FullName?.StartsWith("WorkflowCore") ?? false);
```

Comment:
is this compatible with existing databases out there that have done it the old way?

Reply:
Yes, it can serialize/deserialize in both ways. The new or replaced documents will have the new representation and the old ones will remain as they are.